### PR TITLE
fix(insights): HogQL calculation of saved legacy insights v3

### PR DESCRIPTION
**Pull request by Twixes**
Revert of #21718. See [this internal thread](https://posthog.slack.com/archives/C0368RPHLQH/p1713779182882649) for context.
See commit trail for changes. The important thing is that E2E tests now use HogQL insights, so we'll be catching issues in what's actually relevant.

### CODE REVIEW

File: posthog/hogql_queries/query_runner.py [hunk 1]
```diff
@@ -63,11 +63,11 @@
...
 
 class ExecutionMode(IntEnum):
-    CALCULATION_REQUESTED = 2
+    CALCULATION_ALWAYS = 2
     """Always recalculate."""
-    CALCULATION_ONLY_IF_STALE = 1
+    RECENT_CACHE_CALCULATE_IF_STALE = 1
     """Use cache, unless the results are missing or stale."""
-    CACHE_ONLY = 0
+    CACHE_ONLY_NEVER_CALCULATE = 0
```
Comment (webjunkie) [inline posthog/hogql_queries/query_runner.py:70]: 😄

**Review by webjunkie — APPROVED** [review]

File: posthog/hogql_queries/legacy_compatibility/feature_flag.py [hunk 2]
```diff
@@ -1,26 +1,39 @@
...
-        if not hasattr(user, "distinct_id"):  # exclude api endpoints that don't have auth from the flag
-            return False
+    if posthoganalytics.feature_enabled(
+        GLOBAL_FLAG,
+        user.distinct_id,
+        person_properties={"email": user.email},
+        only_evaluate_locally=True,
+        send_feature_flag_events=False,
+    ):
+        # HogQL insights enabled all the way
```
Comment (webjunkie) [inline posthog/hogql_queries/legacy_compatibility/feature_flag.py:30]: I only now understood how this is intended to work 😅 

I was checking if the queries actually use HogQL... so far I don't run into this condition even with all flags enabled, which means I don't get HogQL results. What could be wrong? The flags are enabled and set to release 100%.

Comment (Twixes) [reply posthog/hogql_queries/legacy_compatibility/feature_flag.py:30]: You should be running this branch with the `hogql-insights-preview` enabled, meaning everything runs on HogQL – definitely works on my machine, hmm

Comment (webjunkie) [reply posthog/hogql_queries/legacy_compatibility/feature_flag.py:30]: I think I double checked... will check again 🏃🏻‍♂️

Comment (webjunkie) [reply posthog/hogql_queries/legacy_compatibility/feature_flag.py:30]: After stepping through with debugger, problem for me is `only_evaluate_locally=True` makes it not get any flag and returns as if it's off for me. When I remove this argument, it works.

Comment (neilkakkar) [reply posthog/hogql_queries/legacy_compatibility/feature_flag.py:30]: (need to run locally with a personal API key for local eval to work)
like:
```
POSTHOG_PERSONAL_API_KEY=phx_n79cT52OfsxAWDhZs9j3w67aRoBCZ7l5ksRRKmAi5nr DEBUG=1  ./bin/start
```

Comment (webjunkie) [reply posthog/hogql_queries/legacy_compatibility/feature_flag.py:30]: TIL

**Pull request merged**